File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    answer = []
    correct = 0 
    for i in win_nums :
        if i in lottos :
            correct+= 1
    zero_count = lottos.count(0)
    
    mincorrect = 0
    maxcorrect = 0 
    if zero_count != 0 :
        mincorrect = correct 
        maxcorrect = correct +zero_count
    else :
        mincorrect = correct 
        maxcorrect = correct
    
    maxrank = 0 
    minrank = 0
    
    if maxcorrect == 6 :
        maxrank = 1
    elif maxcorrect == 5 :
        maxrank = 2
    elif maxcorrect == 4 :
        maxrank = 3
    elif maxcorrect == 3 :
        maxrank = 4
    elif maxcorrect == 2 :
        maxrank = 5
    else :
        maxrank = 6
        
    if mincorrect == 6 :
        minrank = 1
    elif mincorrect == 5 :
        minrank = 2
    elif mincorrect == 4 :
        minrank = 3
    elif mincorrect == 3 :
        minrank = 4
    elif mincorrect == 2 :
        minrank = 5
    else :
        minrank = 6
        
    return [maxrank,minrank]

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int [2] ;
        int zerocnt = 0 ;
        int correctcnt = 0 ;
        boolean [] correct = new boolean[win_nums.length];
        for (int i = 0 ; i<lottos.length; i++)
        {
            if (checkidx(lottos[i],win_nums) != -1)
            {
                //안에있다면
                correct[checkidx(lottos[i],win_nums)] = true;
                correctcnt+=1;
            } else if (lottos[i] == 0)
            {
                zerocnt +=1;
            }
        }
        
        answer[0] = zerocnt == 0 && correctcnt ==0? 6 : (7-correctcnt-zerocnt);
        answer[1] = correctcnt == 0? 6 : (7-correctcnt) ;
        
        
        return answer;
    }
    
    public int checkidx (int val , int [] win_nums)
    {
        for (int i = 0 ; i<win_nums.length; i++)
        {
            if (win_nums[i] == val)
            {
                return i;
            }
        }
        return -1;
    }
}